### PR TITLE
Interface docs update

### LINKED ISSUE #2607: Interface docs need updating
The [docs for the interface submodels](https://pybamm.readthedocs.io/en/latest/source/models/submodels/interface/index.html) should match the folder structure [here](https://github.com/pybamm-team/PyBaMM/tree/develop/pybamm/models/submodels/interface).

### PR DESCRIPTION
**Pull request by dikwickley**
# Description

Updated the docs for the interface module.

Fixes #2607 

## Type of change
documentation update

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by dikwickley** [comment]
@tinosulzer some of these checks failed, any idea what might be causing them cause I didn’t make any changes to code.

**Comment by rtimms** [comment]
@dikwickley can you merge develop? the failing tests aren’t related to your changes

**Comment by tinosulzer** [comment]
@dikwickley any progress on this?

**Review by rtimms — APPROVED** [review]
thanks! looks like there is one error in the docs related to the full utilisation model, but then it’s good to go

**Comment by rtimms** [comment]
@all-contributors please add @dikwickley  for docs

File: docs/source/api/models/submodels/interface/interface_utilisation/index.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Interface Utilisation
+=====================
+
+.. toctree::
+
+ base_utilisation
+ constant_utilisation
+ current_driven_utilisation
+ full_utlisiation
```
Comment (Saransh-cpp) [inline docs/source/api/models/submodels/interface/interface_utilisation/index.rst:9]:
```suggestion
 full_utilisation
```

**Comment by Saransh-cpp** [comment]
Should be good to go now!

**Comment by Saransh-cpp** [comment]
(fixed some doctests)

**Pull request merged**